tornado/httputil.py

```diff
@@ -230,13 +239,127 @@ def parse(cls, headers):
            mix of `KeyError`, and `ValueError`.
 
         """
-        h = cls()
+        h = HTTPHeaders()
         for line in headers.splitlines():
             if line:
                 h.parse_line(line)
         return h
 
-    # MutableMapping abstract method implementations.
+    def pop(self, key, default=__marker):
+        '''D.pop(k[,d]) -> v, remove specified key and return the corresponding value.
+          If key is not found, d is returned if given, otherwise KeyError is raised.
+        '''
+        try:
+            value = self._dict[_normalized_headers[key]]
+        except KeyError:
+            if default is __marker:
+                raise
+            return default
+        else:
+            del self[key]
+            return value
+
+    def popitem(self):
+        '''D.popitem() -> (k, v), remove and return some (key, value) pair
+           as a 2-tuple; but raise KeyError if D is empty.
+        '''
+        try:
+            key = next(iter(self._dict))
+        except StopIteration:
+            raise KeyError
+        value = self.pop(key)
+        return key, value
+
+    def clear(self):
+        'D.clear() -> None.  Remove all items from D.'
+        self._dict.clear()
+        self._as_list.clear()
+        self._last_key = None
+
+    def update(*args, **kwds):
```

Comment:
Why not `update(self, other = ()`?

Reply:
That's a very good question. I copied this off of MutableDict.